include/zephyr/drivers/battery.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_BATTERY
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_BATTERY_H_
+#define ZEPHYR_INCLUDE_DRIVERS_BATTERY_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <zephyr/device.h>
+
+enum fuel_gauge_property_type {
+
+	/* Keep these alphabetized wrt property name */
+
+	/** Runtime Dynamic Battery Parameters */
+	/**
+	 * Provide a 1 minute average of the current on the battery.
+	 * Does not check for flags or whether those values are bad readings.
+	 * See driver instance header for details on implementation and
+	 * how the average is calculated. Units in mA negative=discharging
+	 */
+	FUEL_GAUGE_AVG_CURRENT = 0,
+
+	/** Battery current (mA); negative=discharging */
+	FUEL_GAUGE_CURRENT,
+	/** Whether the battery underlying the fuel-gauge is cut off from charge */
+	FUEL_GAUGE_CHARGE_CUTOFF,
+	/** Cycle count in 1/100ths (number of charge/discharge cycles) */
+	FUEL_GAUGE_CYCLE_COUNT,
+	/** Connect state of battery */
+	FUEL_GAUGE_CONNECT_STATE,
+	/** General Error/Runtime Flags */
+	FUEL_GAUGE_FLAGS,
+	/** Full Charge Capacity in mAh (might change in some implementations to determine wear) */
+	FUEL_GAUGE_FULL_CHARGE_CAPACITY,
+	/** Is the battery physically present */
+	FUEL_GAUGE_PRESENT_STATE,
+	/** Remaining capacity in mAh */
+	FUEL_GAUGE_REMAINING_CAPACITY,
+	/** Remaining battery life time in minutes */
+	FUEL_GAUGE_RUNTIME_TO_EMPTY,
+	/** Remaining time in minutes until battery reaches full charge */
+	FUEL_GAUGE_RUNTIME_TO_FULL,
+	/** Absolute state of charge (percent, 0-100) - expressed as % of design capacity */
+	FUEL_GAUGE_STATE_OF_CHARGE,
+	/** Temperature in 0.1 K */
+	FUEL_GAUGE_TEMPERATURE,
+	/** Battery voltage (mV) */
+	FUEL_GAUGE_VOLTAGE,
+
+	/** Reserved to demark downstream custom properties */
+	FUEL_GAUGE_CUSTOM_BEGIN = 256,
```

Comment:
I think that if you do this the enum will always be 2 bytes. Anyway, how about doing it like the sensor API instead?

https://github.com/zephyrproject-rtos/zephyr/blob/33e583b6437e460006e20d04ef822b0954edb586/include/zephyr/drivers/sensor.h#L188-L197

That way there's only one gap.